torchtext/data/functional.py

```diff
@@ -151,3 +153,54 @@ def simple_space_split(iterator):
 
     for line in iterator:
         yield line.split()
+
+
+def read_text_iterator(path, tokenizer):
+    r"""Read text from path and yield a list of tokens based on the tokenizer
+
+    Arguments:
+        path: the file path.
+        tokenizer: the tokenizer used to tokenize string text.
+
+    Examples:
+        >>> from torchtext.data.functional import read_text_iterator
+        >>> tokenizer = get_tokenizer("basic_english")
+        >>> list((read_text_iterator('.data/ptb.train.txt', tokenizer)))
+            [['Sentencepiece', 'encode', 'as', 'pieces'], ['example', 'to', 'try!']]
+    """
+
+    with io.open(path, encoding="utf8") as f:
+        reader = unicode_csv_reader(f)
```

Comment:
What if it's not a csv?

Reply:
Following this line, it calls `join` func before sending to `tokenizer`. Therefore, it works fine for non-csv files.
```
tokens = tokenizer(' '.join(row))
```

Reply:
For non-CSV files we might care about the "," as an explicit datapointer. So, for example, in a pure-text setting the comma might have a meaning, but in this case it gets replaced with a space without the user knowing about it, because the csv reader will interpret it as a boundary.

Reply:
This doesn't work.

Reply:
I don't think there is much value in this function as an abstraction. It wraps a one-liner

```iter(tokenizer(row) for row in io.open(path, encoding="utf8")``` once the incorrect usage of the unicode_csv reader has been removed.